File: modul_praktikum_python/contoh_code/python/daf_pegawai_array.py
"""Daftar 5 pegawai: input lalu cetak satu per satu."""

from dataclasses import dataclass


@dataclass
class Pegawai:
    nama: str = ""
    gol: str = " "
    jml_anak: int = 0


def input_pegawai(pegawai: Pegawai) -> None:
    pegawai.nama = input("Nama        : ")
    pegawai.gol = input("Gol (A - D) : ").strip().upper()[:1]
    pegawai.jml_anak = int(input("Jumlah Anak : "))


def upah_kotor(gol: str) -> float:
    g = gol.upper()
    if g == "A":
        return 1_000_000.0
    if g == "B":
        return 800_000.0
    if g == "C":
        return 600_000.0
    if g == "D":
        return 400_000.0
    return 0.0


def persen_tunjangan(jml_anak: int) -> float:
    if jml_anak > 2:
        return 0.3
    return 0.2


def cetak(pegawai: Pegawai) -> None:
    print("Nama        :", pegawai.nama)
    print("Gol (A - D) :", pegawai.gol)
    print("Jumlah Anak :", pegawai.jml_anak)
    uk = upah_kotor(pegawai.gol)
    upah_bersih = uk - (uk * persen_tunjangan(pegawai.jml_anak))
    print(f"Upah        : {upah_bersih:5.2f}")


def main() -> None:
    daf_pegawai = [Pegawai() for _ in range(5)]
    for i in range(5):
        print("Data ke ", i + 1, ":", sep="")
        input_pegawai(daf_pegawai[i])
    for i in range(5):
        print("Data ke ", i + 1, ":", sep="")
        cetak(daf_pegawai[i])
        input("Tekan Enter untuk melanjutkan...")


if __name__ == "__main__":
    main()

File: modul_praktikum_python/contoh_code/pascal/daf_pegawai_array.pas
uses crt; 
type 
  Tpegawai=record 
           Nama : string[20]; 
           Gol  : char; 
           JmlAnak : integer; 
           end; 
  Tdaf_peg=array[1..5]of Tpegawai; 

procedure input(var pegawai:tpegawai); 
begin 
 with pegawai do 
 begin  
  write(‘Nama        : ’);readln(Nama); 
  write(‘Gol (A - D) : ’);readln(Gol); 
  write(‘Jumlah Anak : ’);readln(JmlAnak); 
 end; 
end; 

function upahkotor(gol:char):real; 
begin 
  Case Gol of 
  ‘A’ : upahkotor:=1000000; 
  ‘B’ : upahkotor:=800000; 
  ‘C’ : upahkotor:=600000; 
  ‘D’ : upahkotor:=400000; 
  end;   
end; 

function persentunjangan(jmlanak:integer):real; 
begin 
  if(JmlAnak>2)then  
    persenTunjangan := 0.3   
  else 
   persenTunjangan := 0.2; 
end; 

procedure cetak(pegawai:tpegawai); 
var 
upahbersih : real; 
begin 
 clrscr; 
 with pegawai do 
 begin  
  writeln(‘Nama        : ’,Nama); 
  writeln(‘Gol (A - D) : ’,Gol); 
  writeln(‘Jumlah Anak : ’,JmlAnak); 
  upahbersih:= upahkotor(gol) – 
(upahkotor(gol)*persentunjangan(jmlanak)); 
  writeln(‘Upah        : ’,upahbersih:5:2); 
 end; 
end; 

var 
  daf_pegawai:tdaf_peg; 
  i:integer; 
Begin 
For i:=1 to 5 do   
begin 
  writeln('Data ke ',i,':'); 
  input(daf_pegawai[i]); 
end; 
For i:=1 to 5 do   
 begin 
   writeln('Data ke ',i,':'); 
   cetak(daf_pegawai[i]); 
   readkey; 
 end; 
end.
